README.md
# language-detection
| Build Status | Code Coverage | Version | Total Downloads | Minimum PHP Version | License |
| :---: | :---: | :---: | :---: | :---: | :---: |
| [![Build Status](https://travis-ci.org/patrickschur/language-detection.svg?branch=master)](https://travis-ci.org/patrickschur/language-detection) | [![codecov](https://codecov.io/gh/patrickschur/language-detection/branch/master/graph/badge.svg)](https://codecov.io/gh/patrickschur/language-detection) | [![Version](https://img.shields.io/packagist/v/patrickschur/language-detection.svg?style=flat-square)](https://packagist.org/packages/patrickschur/language-detection) | [![Total Downloads](https://img.shields.io/packagist/dt/patrickschur/language-detection.svg?style=flat-square)](https://packagist.org/packages/patrickschur/language-detection) | [![Minimum PHP Version](https://img.shields.io/badge/php-%3E%3D%207.3-4AC51C.svg?style=flat-square)](https://php.net/) | [![License](https://img.shields.io/packagist/l/patrickschur/language-detection.svg?style=flat-square)](https://opensource.org/licenses/MIT) |

This library can detect the language of a given text string.
It can parse given training text in many different idioms into a sequence of [N-grams](https://en.wikipedia.org/wiki/N-gram) and builds a database file in PHP to be used in the detection phase.
Then it can take a given text and detect its language using the database previously generated in the training phase.
The library comes with text samples used for training and detecting text in 110 languages.

## Table of Contents
- [Installation with Composer](#installation-with-composer)
- [How to upgrade from 3.y.z to 4.y.z?](#how-to-upgrade)
- [Basic Usage](#basic-usage)
- [API](#api)
- [Method Chaining](#method-chaining)
- [Array Access](#arrayaccess)
- [List of supported languages](#supported-languages)
- [Other languages](#other-languages)
- [FAQ](#faq)
- [Contributing](#contributing)
- [License](#license)

## Installation with Composer
> **Note:** This library requires the [Multibyte String](https://secure.php.net/manual/en/book.mbstring.php) extension in order to work. 

```bash
$ composer require patrickschur/language-detection
```

## <a name="how-to-upgrade"></a> How to upgrade from `3.y.z` to `4.y.z`?

**Important**: Only for people who are using a **custom directory** with their **own** translation files.

Starting with version `4.y.z` we have updated the resource files. For performance reasons we now use PHP instead of JSON as a format. That means people who want to use `4.y.z` and used `3.y.z` before, have to upgrade their JSON files to PHP. To upgrade your resource files you must generate a language profile again. The JSON files are then no longer needed.

You can delete unnecessary JSON files under Linux with the following command.

```sh
rm resources/*/*.json
```

## Basic Usage
To detect the language correctly, the length of the input text should be at least some sentences.
```php
use LanguageDetection\Language;
 
$ld = new Language;
 
$ld->detect('Mag het een onsje meer zijn?')->close();
```
Result:
```text
Array
(
    "nl" => 0.66193548387097,
    "af" => 0.51338709677419,
    "br" => 0.49634408602151,
    "nb" => 0.48849462365591,
    "nn" => 0.48741935483871,
    "fy" => 0.47822580645161,
    "dk" => 0.47172043010753,
    "sv" => 0.46408602150538,
    "bi" => 0.46021505376344,
    "de" => 0.45903225806452,
    [...]
)
```

## API

#### `__construct(array $result = [], string $dirname = '')`
You can pass an array of languages to the constructor. To compare the desired sentence only with the given languages.
This can dramatically increase the performance. 
The other parameter is optional and the name of the directory where the translations files are located.
```php

CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.

## [5.0.0] - 2020-12-11
### Added
- Support for PHP 8
- Swahili and Oromo language files (thanks to @matthewnessworthy)
### Removed
- PHP 7.2 support as it is no longer supported

## [4.0.1] - 2020-08-12
### Changed
- Fixed Unsupported operand types for custom directory (thanks to @iquito)

## [4.0.0] - 2020-08-09
### Changed
- Converted all resource files to PHP for performance reasons (thanks to @Pierstoval)

## [3.4.2] - 2019-11-16
### Changed
- Removed a build failure by using a newer version of PHPUnit

## [3.4.1] - 2019-11-16
### Changed
- Optimized performance by using fully-qualified function calls, thanks to [draco2003](https://github.com/draco2003)
- PHP 7.2 or higher is now used for testing
- Removed output, see issue [#22](https://github.com/patrickschur/language-detection/issues/22)

## [3.4.0] - 2018-09-19
### Changed
- Danish language code, thanks to [bbalet](https://github.com/bbalet)

## [3.3.0] - 2018-06-07
### Added
- Serbian cyrylic and serbian latin language files

### Changed
- The method `__toString` returns always a string

## [3.2.0] - 2017-08-20
### Added
- Experimental support for custom directories. You can now specify your own path for translation files.
- Credits

### Changed
- Corrections to Amharic (`am`) translation
- Corrections to Czech (`cs`) translation
- Corrections to Lao (`lo`) translation

## [3.1.1] - 2017-07-27
### Changed
- Return empty object if no result was found

## [3.1.0] - 2017-03-14
### Added
- Support for the languages Lingala, Sanskrit, Tonga and Urdu

## [3.0.0] - 2017-02-05
### Added
- `CHANGELOG.md`
- `CONTRIBUTING.md`
- Tokenizer interface

### Changed
- Folder structure (renamed `etc` to `resources`)
- Renamed all language files (added **.txt** extension)
- Improved performance

resources/ay/ay.php
<?php

return array (
  'ay' => 
  array (
    0 => 'a',
    1 => 'i',
    2 => 'k',
    3 => 't',
    4 => 'n',
    5 => 's',
    6 => 'h',
    7 => 'p',
    8 => 'a_',
    9 => 'j',
    10 => 'u',
    11 => 'ka',
    12 => 'm',
    13 => 'an',
    14 => 'pa',
    15 => 'ak',
    16 => 'na',
    17 => 'at',
    18 => 'r',
    19 => 'q',
    20 => 'ha',
    21 => 'ta',
    22 => 'ap',
    23 => 'as',
    24 => 'w',
    25 => 'ñ',
    26 => 'apa',
    27 => 'ma',
    28 => 'ar',
    29 => 'aq',
    30 => 'ña',
    31 => 'ja',
    32 => '_j',
    33 => '_m',
    34 => 'ch',
    35 => 'i_',
    36 => 'nak',
    37 => 'aka',
    38 => 'c',
    39 => 'y',
    40 => 'jh',
    41 => 'e',
    42 => '_ja',
    43 => '_a',
    44 => 'si',
    45 => 'ata',
    46 => '_ma',
    47 => 'ki',
    48 => 'w_',
    49 => 'am',
    50 => 'is',
    51 => 'ana',
    52 => 'aña',
    53 => 'añ',
    54 => 'asi',
    55 => 'qe',
    56 => 'l',
    57 => '_t',
    58 => 'aqe',
    59 => 'aki',
    60 => 'ti',
    61 => 'cha',
    62 => 'in',
    63 => 'ñap',
    64 => '_u',
    65 => 'qa',
    66 => '\'',
    67 => 's_',
    68 => 'ni',
    69 => '\'a',
    70 => 'ay',
    71 => 'aw',
    72 => 'jha',
    73 => 'ru',
    74 => 'sa',
    75 => '_k',
    76 => 'mar',
    77 => 'wa',
    78 => 'ya',
    79 => 'un',
    80 => 'aw_',
    81 => 'kan',
    82 => 'it',
    83 => 'ark',
    84 => '_ch',
    85 => 'aru',
    86 => 'una',
    87 => 'uk',
    88 => 'h_',
    89 => 'ti_',
    90 => 'jaq',
    91 => 'paw',
    92 => 'pat',
    93 => 'rk',
    94 => 'jh_',
    95 => '_ar',
    96 => 'rka',
    97 => 'hat',
    98 => 'tak',
    99 => '_ta',
    100 => '_c',
    101 => 'ach',
    102 => 'ama',
    103 => '_wa',
    104 => 'iw_',
    105 => 'kh',
    106 => 'n_',
    107 => 'iw',
    108 => 'aqa',
    109 => 'ani',
    110 => 'kap',
    111 => 'ki_',
    112 => 'na_',
    113 => 'spa',
    114 => 't\'a',
    115 => 'taq',
    116 => 'sp',
    117 => 'ac',
    118 => '_w',
    119 => 'ir',
    120 => 'mu',
    121 => 't_',
    122 => 't\'',
    123 => 'jan',
    124 => 'sa_',
    125 => 'pa_',
    126 => 'e_',
    127 => 'qe_',
    128 => '_uk',
    129 => 'ra',
    130 => 'kas',
    131 => 'kis',
    132 => 'pi',
    133 => 'ha_',
    134 => 'qa_',
    135 => 'at_',
    136 => 'niw',
    137 => 'may',
    138 => 'ina',
    139 => '_y',
    140 => 'ä',
    141 => 'ati',
    142 => 'pan',
    143 => '_kh',
    144 => 'o',
    145 => 'hi',
    146 => '_ya',
    147 => '\'aq',
    148 => 'aj',
    149 => 'mp',
    150 => '_i',
    151 => '_mu',
    152 => '_t\'',
    153 => 'iti',
    154 => 'ayn',
    155 => 'ka_',
    156 => 'u_',
    157 => 'us',
    158 => 'as_',
    159 => 'amp',
    160 => 'pj',
    161 => 'ip',
    162 => 'pjh',
    163 => 'an_',
    164 => 'mun',
    165 => 'ru_',
    166 => 'yni',
    167 => 'st',
    168 => 'ut',
    169 => 'yn',
    170 => 'iña',
    171 => 'ajh',
    172 => 'uka',
    173 => 'li',
    174 => 'iñ',
    175 => 'hit',
    176 => 'tä',
    177 => 'oq',
    178 => 'ñan',
    179 => 'is_',
    180 => 'isp',
    181 => 'oqa',
    182 => 'ejh',
    183 => 'qen',
    184 => 'ch\'',
    185 => 'khi',
    186 => 'isi',
    187 => 'has',
    188 => '_s',
    189 => 'atä',
    190 => 'nch',
    191 => 'kañ',
    192 => 'sis',
    193 => 'p_',
    194 => 'll',
    195 => 'sk',
    196 => 'rus',
    197 => 'kam',
    198 => 'mpi',
    199 => 'han',
    200 => 'siñ',
    201 => 'qha',
    202 => 'en',
    203 => 'uñ',
    204 => 'ej',
    205 => 'h\'',
    206 => 'al',
    207 => '_am',
    208 => 'asp',
    209 => 'sin',
    210 => 'h\'a',
    211 => 'ham',
    212 => '_in',
    213 => 'qat',
    214 => 'ñat',
    215 => 'nc',
    216 => 'nk',
    217 => 'qh',
    218 => 'äña',
    219 => '_sa',
    220 => 'nka',
    221 => 'hañ',
    222 => 'yas',
    223 => '_uñ',
    224 => 'yat',
    225 => 'ita',
    226 => 'ku',
    227 => 'rj',
    228 => 'lu',
    229 => 'ns',
    230 => 'arj',
    231 => 'ska',
    232 => 'tap',
    233 => 'sti',
    234 => 'kha',
    235 => 'asa',
    236 => 'täñ',
    237 => 'ma_',
    238 => 'sit',
    239 => 'tha',
    240 => 'ta_',
    241 => 'tas',
    242 => 'ri',
    243 => 'ji',
    244 => 'tj',
    245 => '_q',
    246 => 'iri',
    247 => 'sna',
    248 => '_ji',
    249 => 'ara',
    250 => 'äñ',
    251 => 'he',
    252 => 'uy',
    253 => 'la',
    254 => 'th',
    255 => '_p',
    256 => 'ur',
    257 => 'tis',
    258 => 'jam',
    259 => 'tat',
    260 => '_ut',
    261 => 'pas',
    262 => 'ñja',
    263 => '_ku',
    264 => 'apj',
    265 => 'pi_',
    266 => 'kat',
    267 => 'rjh',
    268 => 'hap',
    269 => 'noq',
    270 => '_l',
    271 => 'sn',
    272 => '_ju',
    273 => 'asn',
    274 => 'sip',
    275 => '_ka',
    276 => 'uñj',
    277 => 'qas',
    278 => 'ura',
    279 => 'aya',
    280 => 'ans',
    281 => 'nañ',
    282 => 'kaj',
    283 => 'nin',
    284 => 'ukh',
    285 => 'il',
    286 => 'no',
    287 => 'ñj',
    288 => 'eq',
    289 => 'us_',
    290 => 'kun',
    291 => 'anc',
    292 => 'pac',
    293 => 'ino',
    294 => 'api',
    295 => 'nj',
    296 => 'ju',
    297 => 'tan',
    298 => 'jil',
    299 => 'lir',
    300 => 'uya',
    301 => '\'am',
    302 => 'ili',
    303 => 'ña_',
    304 => 'utj',
    305 => 'ik',
    306 => 'hu',
    307 => '\'as',
    308 => 'chi',
    309 => 'ink',
    310 => 'tañ',
    311 => 'ali',
    312 => '_lu',
    313 => 'kak',
    314 => 'rak',
    315 => 'ipa',
    316 => 'muy',
    317 => 'isa',
    318 => 'mas',
    319 => 'pis',
    320 => 'njh',
    321 => 'hac',
    322 => 'kiw',
    323 => 'amu',
    324 => 'up',
    325 => 'jas',
    326 => 'in_',
    327 => 'upa',
    328 => 'nan',
    329 => 'nsa',
    330 => 'ena',
    331 => 'wa_',
    332 => 'tay',
    333 => 'wak',
    334 => 'way',
    335 => '\'a_',
    336 => 'yt',
    337 => 'mi',
    338 => 'rin',
    339 => 'anq',
    340 => 'heq',
    341 => 'che',
    342 => 'iru',
    343 => 'lan',
    344 => 'mp_',
    345 => 'ank',
    346 => 'khu',
    347 => 'en_',
    348 => 'lur',
    349 => 'rañ',
    350 => 'wal',
    351 => 'lin',
    352 => 'ipj',
    353 => 'mat',
    354 => 'eqa',
    355 => 'ayt',
    356 => '_\'',
    357 => 'llu',
    358 => 'run',
    359 => 'qpa',
    360 => 'yaq',
    361 => 'awa',
    362 => 'pam',
    363 => '_ay',
    364 => 'yll',
    365 => 'tja',
    366 => 'kar',
    367 => 'ayl',
    368 => 'sap',
    369 => 'uki',
    370 => 'qej',
    371 => 'usk',
    372 => 'anj',
    373 => 'aqh',
    374 => 'nip',
    375 => 'au',
    376 => 'k\'',
    377 => 'nq',
    378 => 'qp',
    379 => 'su',
    380 => 'yl',
    381 => 'nt',
    382 => 'ñ_',
    383 => 'mä',
    384 => 'j_',
    385 => 'tu',
    386 => 'wi',
    387 => 'k_',
    388 => 'nap',
    389 => 'tir',
    390 => 'inc',
    391 => 'ast',
    392 => 'naq',
    393 => 'ath',
    394 => 'aqp',
    395 => 'jhe',
    396 => 'uch',
    397 => 'sar',
    398 => 'jac',
    399 => '_pa',
    400 => 'ayk',
    401 => '_ak',
    402 => 'hus',
    403 => 'ap_',
    404 => 'isk',
    405 => 'ni_',
    406 => 'k\'a',
    407 => 'ak_',
    408 => 'r_',
    409 => 'uc',
    410 => 'rm',
    411 => 'um',
    412 => 'yk',
    413 => 'nir',
    414 => 'ri_',
    415 => 'nqh',
    416 => 'ch_',
    417 => 'sir',
    418 => 'hik',
    419 => 'arm',
    420 => 'mac',
    421 => 'ir_',
    422 => 'yt\'',
    423 => 'ika',
    424 => 'ns_',
    425 => 'ask',
    426 => 'man',
    427 => 'qer',
    428 => 'ust',
    429 => '_un',
    430 => 'juc',
    431 => 'ist',
    432 => 'war',
    433 => 'auk',
    434 => 'pit',
    435 => 'ï',
    436 => 'ph',
    437 => '\'u',
    438 => 'er',
    439 => 'ho',
    440 => 'yan',
    441 => 'ñas',
    442 => 'sth',
    443 => 'par',
    444 => 'kjh',
    445 => '_aj',
    446 => 'mä_',
    447 => 'si_',
    448 => 'pho',
    449 => 'ala',
    450 => 'hoq',
    451 => '_mä',
    452 => 'kit',
    453 => 'win',
    454 => '_ph',
    455 => 'lla',
    456 => '_\'a',
    457 => '_ña',
    458 => 'qañ',
    459 => 'nas',
    460 => 'pka',
    461 => 'jañ',
    462 => 'rmi',
    463 => 'mis',
    464 => 'kj',
    465 => 'ps',
    466 => 'pk',
    467 => '_ñ',
    468 => 'ä_',
    469 => 'qo',
    470 => 'ul',
    471 => 'awi',
    472 => 'atj',
    473 => 'utu',
    474 => 'ipk',
    475 => 'kau',
    476 => 'ant',
    477 => 'ñar',
    478 => 'wan',
    479 => '_qh',
    480 => 'rat',
    481 => 'kir',
    482 => '_is',
    483 => '_qe',
    484 => 'ps_',
    485 => 'ins',
    486 => 'añ_',
    487 => 'mta',
    488 => 'ull',
    489 => 'aps',
    490 => 'he_',
    491 => 'inj',
    492 => 'jup',
    493 => 'uni',
    494 => 'ich',
    495 => 'amt',
    496 => 'eta',
    497 => 'aqj',
    498 => 'qet',
    499 => '_ap',
    500 => 'uta',
    501 => 'it_',
    502 => 'waw',
    503 => 'äk',
    504 => 'te',
    505 => '\'e',
    506 => 'ic',
    507 => 'im',
    508 => 'pt',
    509 => 'qj',
    510 => 'et',
    511 => 'mt',
    512 => 'í',
    513 => '_la',
    514 => 'kip',
    515 => '_qo',
    516 => 'paj',
    517 => '_as',
    518 => 'taw',
    519 => 'ats',
    520 => 'nta',
    521 => 'iwi',
    522 => '_yä',
    523 => 'tic',
    524 => 'paq',
    525 => 'täp',
    526 => 'raq',
    527 => 'äpj',
    528 => '_al',
    529 => 'toq',
    530 => 'pja',
    531 => 'nti',
    532 => 'jat',
    533 => 'imp',
    534 => 'qes',
    535 => 'kut',
    536 => 'tin',
    537 => 'har',
    538 => 'qjh',
    539 => 'rup',
    540 => '_j_',
    541 => 'ira',
    542 => 'maw',
    543 => 'ikj',
    544 => 'jik',
    545 => 'qel',
    546 => '_su',
    547 => 'jar',
    548 => 'ras',
    549 => 'uma',
    550 => 'ïña',
    551 => 'yka',
    552 => 'hak',
    553 => 'huy',
    554 => 'uña',
    555 => '_qa',
    556 => 'mut',
    557 => 'pin',
    558 => 'akï',
    559 => 'tiñ',
    560 => '_n',
    561 => 'ïñ',
    562 => 'es',
    563 => 'q\'',
    564 => 'kï',
    565 => 'lq',
    566 => 'el',
    567 => 'ñs',
    568 => 'ts',
    569 => '_h',
    570 => 'or',
    571 => 'kt',
    572 => 'äp',
    573 => 'än',
    574 => 'oj',
    575 => 'yä',
    576 => 'to',
    577 => 'nï',
    578 => 'iy',
    579 => 'uti',
    580 => 'san',
    581 => 'rkn',
    582 => 'tru',
    583 => 'nay',
    584 => 'uk\'',
    585 => '\'ul',
    586 => '_iu',
    587 => 'ruk',
    588 => '_tr',
    589 => 'la_',
    590 => 'rej',
    591 => 'siw',
    592 => 'juk',
    593 => 'kna',
    594 => 'ipt',
    595 => 'iur',
    596 => 'ire',
    597 => 'lli',
    598 => 'nst',
    599 => 'oqe',
    600 => 'eru',
    601 => 'yt_',
    602 => 'yma',
    603 => 'chu',
    604 => 'kt\'',
    605 => 'hew',
    606 => 'ra_',
    607 => 'ajl',
    608 => 'jak',
    609 => 'sk\'',
    610 => 'rja',
    611 => 'jal',
    612 => '_us',
    613 => 'iya',
    614 => '_ti',
    615 => 'lun',
    616 => 'nat',
    617 => 'äki',
    618 => '_mi',
    619 => '_\'u',
    620 => 'nit',
    621 => 'tji',
    622 => 'nja',
    623 => 'tuñ',
    624 => 'ji_',
    625 => '_au',
    626 => 'lup',
    627 => 'sta',
    628 => 'akh',
    629 => 'iñ_',
    630 => 'lqa',
    631 => 'unt',
    632 => 'iki',
    633 => 'am_',
    634 => '_th',
    635 => 'jch',
    636 => 'ell',
    637 => 'teq',
    638 => 'yac',
    639 => 'roj',
    640 => 'waj',
    641 => 'nej',
    642 => 'sp_',
    643 => 'ajc',
    644 => '\'aj',
    645 => '\'eq',
    646 => 'ojh',
    647 => 'ate',
    648 => 'yañ',
    649 => 'itä',
    650 => '_ha',
    651 => 'qal',
    652 => 'rna',
    653 => 'ne',
    654 => 'jc',
    655 => 'ym',
    656 => 'tr',
    657 => 'ks',
    658 => 'm_',
    659 => 'ep',
    660 => 'rn',
    661 => 'jl',
    662 => 'ro',
    663 => 'iu',
    664 => 'uj',
    665 => 'ew',
    666 => 're',
    667 => 'kn',
    668 => 'pej',
    669 => 'mpe',
    670 => 'amj',
    671 => '_ki',
    672 => 'wej',
    673 => 'uyt',
    674 => 'hay',
    675 => 'yta',
    676 => 'nki',
    677 => 'pak',
    678 => 'akt',
    679 => 'uñs',
    680 => 'ñst',
    681 => 'lay',
    682 => '_to',
    683 => '_pi',
    684 => 'äsp',
    685 => 'rim',
    686 => 'qaw',
    687 => 'lis',
    688 => 'rj_',
    689 => 'qol',
    690 => 'utä',
    691 => 'eqe',
    692 => 'muk',
    693 => 'tit',
    694 => 'ite',
    695 => 'pta',
    696 => 'nik',
    697 => '_aq',
    698 => 'äni',
    699 => 'tän',
    700 => 'awe',
    701 => '_si',
    702 => 'tej',
    703 => 'ukt',
    704 => 'nq\'',
    705 => 'irn',
    706 => '_ur',
    707 => '_ir',
    708 => 'mja',
    709 => 'atw',
    710 => 'twa',
    711 => '\'ay',
    712 => 'sam',
    713 => 'raw',
    714 => 'alj',
    715 => 'ew_',
    716 => '_il',
    717 => 'un_',
    718 => 'usu',
    719 => 'es_',
    720 => 'kïñ',
    721 => 'sun',
    722 => 'hwa',
    723 => 'qta',
    724 => 'ya_',
    725 => 'tj_',
    726 => 'utt',
    727 => 'lka',
    728 => 'llk',
    729 => 'sul',
    730 => 'ku_',
    731 => 'ila',
    732 => '_añ',
    733 => 'nïs',
    734 => 'anï',
    735 => 'stu',
    736 => 'all',
    737 => 'aní',
    738 => 'kïk',
    739 => 'ane',
    740 => 'aro',
    741 => 'saw',
    742 => 'amä',
    743 => 'chj',
    744 => 'hja',
    745 => '_nu',
    746 => 'uwa',
    747 => 'ayr',
    748 => 'rut',
    749 => 'jhw',
    750 => 'hir',
    751 => 'llq',
    752 => 'qan',
    753 => 'ump',
    754 => 'eq_',
    755 => 'hsa',
    756 => 'jhs',
    757 => 'kaw',
    758 => 'kie',
    759 => 'tsa',
    760 => 'iej',
    761 => 'yi_',
    762 => 'tuy',
    763 => '\'um',
    764 => 'k\'u',
    765 => 'sum',
    766 => '_k\'',
    767 => 'tjn',
    768 => 'jno',
    769 => 'yra',
    770 => '_na',
    771 => 'qep',
    772 => 'ona',
    773 => 'irj',
    774 => '_t_',
    775 => 'jll',
    776 => 'yna',
    777 => 'ski',
    778 => 'waq',
    779 => 'aqo',
    780 => 'qon',
    781 => 'uym',
    782 => 'tjk',
    783 => '_q\'',
    784 => 'q\'e',
    785 => 'qor',
    786 => 'orp',
    787 => 'rpa',
    788 => 'pap',
    789 => '\'eu',
    790 => 'map',
    791 => 'apt',
    792 => 'rk\'',
    793 => 'ato',
    794 => 'eri',
    795 => 'äna',
    796 => 'tjh',
    797 => 'jta',
    798 => 'lu_',
    799 => 'täk',
    800 => 'äka',
    801 => 'ayp',
    802 => 'ypi',
    803 => 'hip',
    804 => 'yän',
    805 => 'lum',
    806 => 'tiy',
    807 => 'nïñ',
    808 => '_or',
    809 => 'ora',
    810 => 'enï',
    811 => 'aks',
    812 => 'ars',
    813 => 'rsu',
    814 => 'añs',
    815 => 'ñsa',
    816 => 'yku',
    817 => 'we',
    818 => '_o',
    819 => 'hw',
    820 => 'jn',
    821 => 'jk',
    822 => 'lk',
    823 => 'nu',
    824 => 'uw',
    825 => 'yi',
    826 => 'hs',
    827 => 'q_',
    828 => 'yr',
    829 => 'ie',
    830 => 'ïk',
    831 => 'ní',
    832 => 'äs',
    833 => 'lj',
    834 => '\'i',
    835 => 'tw',
    836 => 'rp',
    837 => 'hj',
    838 => 'ík',
    839 => 'on',
    840 => 'tt',
    841 => 'ïs',
    842 => 'qt',
    843 => 'eu',
    844 => 'rs',
    845 => 'pe',
    846 => 'yp',
    847 => 'ol',
    848 => 'mj',
    849 => 'jt',
    850 => 'ë',
    851 => 'yks',
    852 => 'ksn',
    853 => 'sku',
    854 => '_ll',
    855 => 'est',
    856 => 'pur',
    857 => 'wiñ',
    858 => 'lit',
    859 => 'apn',
    860 => 'pna',
    861 => 'aj_',
    862 => 'ja_',
    863 => 'hau',
    864 => 'auq',
    865 => 'uqh',
    866 => 'maj',
    867 => 'liñ',
    868 => 'ñaj',
    869 => 'rk_',
    870 => 'eut',
    871 => 'pt\'',
    872 => 'ptj',
    873 => 'ill',
    874 => 'mil',
    875 => '_pu',
    876 => 'yak',
    877 => 'mak',
    878 => '_\'e',
    879 => 'eps',
    880 => 'arú',
    881 => 'rú_',
    882 => '_yu',
    883 => 'yup',
    884 => 'pay',
    885 => 'ayc',
    886 => 'ych',
    887 => 'kst',
    888 => 'suñ',
    889 => 'ar_',
    890 => 'phi',
    891 => 'qey',
    892 => 'rap',
    893 => 'eya',
    894 => 'ip_',
    895 => 'aph',
    896 => 'yne',
    897 => 'psa',
    898 => 'suk',
    899 => 'qët',
    900 => 'h\'e',
    901 => 'sak',
    902 => 'eqt',
    903 => 'mim',
    904 => 'ljt',
    905 => 'ts_',
    906 => '\'aw',
    907 => 'uru',
    908 => 'mip',
    909 => 'yäc',
    910 => 'say',
    911 => 'urk',
    912 => 'kwa',
    913 => 'akw',
    914 => 'up_',
    915 => 'äch',
    916 => 'sat',
    917 => 'tar',
    918 => 'ul_',
    919 => '_ul',
    920 => 'nth',
    921 => 'akj',
    922 => 'lpa',
    923 => 'llp',
    924 => 'tiw',
    925 => '\'ap',
    926 => 'itu',
    927 => 'iiñ',
    928 => 'esa',
    929 => 'qap',
    930 => 'oqt',
    931 => 'mpa',
    932 => 'tum',
    933 => '_tu',
    934 => 'ujt',
    935 => 'euj',
    936 => 'tka',
    937 => 'tu_',
    938 => 'ptk',
    939 => 'yki',
    940 => 'ini',
    941 => 'ipi',
    942 => '\'añ',
    943 => 'nuk',
    944 => 'h\'u',
    945 => 'ks_',
    946 => 'lii',
    947 => 'qja',
    948 => 'sur',
    949 => 'lqe',
    950 => 'olq',
    951 => '_ac',
    952 => 'kta',
    953 => 'rmu',
    954 => 'urm',
    955 => 'jma',
    956 => 'q\'a',
    957 => 'ijm',
    958 => '_ij',
    959 => 'nic',
    960 => '_im',
    961 => 'wi_',
    962 => 'upj',
    963 => 'mäs',
    964 => 'ïsp',
    965 => 'oll',
    966 => 'jli',
    967 => 'ïka',
    968 => 'hur',
    969 => '\'at',
    970 => 'äqa',
    971 => 'yäq',
    972 => '_uc',
    973 => 'wim',
    974 => '\'uk',
    975 => 'elq',
    976 => 'lañ',
    977 => 'haw',
    978 => 'asu',
    979 => 'min',
    980 => 'wac',
    981 => 'aym',
    982 => 'pañ',
    983 => 'erj',
    984 => 'imi',
    985 => 'ëta',
    986 => 'yoq',
    987 => 'sas',
    988 => '_an',
    989 => 'lut',
    990 => 'piw',
    991 => 'iyi',
    992 => 'ñäk',
    993 => 'añä',
    994 => '_iy',
    995 => 'yau',
    996 => 'au_',
    997 => 'qay',
    998 => 'ajw',
    999 => '\'al',
    1000 => 'qar',
    1001 => 'tst',
    1002 => 'lal',
    1003 => 'epa',
    1004 => 'jay',
    1005 => 'laj',
    1006 => 'ewa',
    1007 => 'níp',
    1008 => 'int',
    1009 => 'jwa',
    1010 => 'ípj',
    1011 => '_má',
    1012 => 'akp',
    1013 => 'kpa',
    1014 => '_yo',
    1015 => 'mäñ',
    1016 => 'iwa',
    1017 => 'itj',
    1018 => 'nís',
    1019 => 'ísi',
    1020 => 'isn',
    1021 => 'má_',
    1022 => 'aha',
    1023 => '_jí',
    1024 => 'ñka',
    1025 => 'añk',
    1026 => 'kch',
    1027 => 'ukc',
    1028 => 'jík',
    1029 => 'íkj',
    1030 => '_mr',
    1031 => 'mra',
    1032 => 'tah',
    1033 => 'st\'',
    1034 => 'naj',
    1035 => 'miñ',
    1036 => 'täs',
    1037 => 'hua',
    1038 => '_hu',
    1039 => 'jap',
    1040 => 'eña',
    1041 => 'qeñ',
    1042 => 'ywa',
    1043 => 'ljh',
    1044 => 'p\'e',
    1045 => '_p\'',
    1046 => 'chh',
    1047 => 't\'u',
    1048 => 'hho',
    1049 => 'ëña',
    1050 => 'tëñ',
    1051 => 'atë',
    1052 => 'li_',
    1053 => 'hoj',
    1054 => 'ojr',
    1055 => 'jri',
    1056 => 'aqë',
    1057 => 'ual',
    1058 => '\'un',
    1059 => 'aja',
    1060 => 'ism',
    1061 => 'q\'o',
    1062 => '\'os',
    1063 => 'os_',
    1064 => 'h\'i',
    1065 => '\'iy',
    1066 => 'yar',
    1067 => 'uku',
    1068 => 'ami',
    1069 => 'smu',
    1070 => 'mir',
    1071 => 'unj',
    1072 => 'ris',
    1073 => 'mpí',
    1074 => 'pík',
    1075 => 'íka',
    1076 => 'ruw',
    1077 => 'nku',
    1078 => 'kus',
    1079 => 'usi',
    1080 => '_uy',
    1081 => 'uyw',
    1082 => 'umi',
    1083 => 'umä',
    1084 => 'apk',
    1085 => 'iha',
    1086 => 'tuk',
    1087 => 'uñ_',
    1088 => 'nuw',
    1089 => 'was',
    1090 => 'päk',
    1091 => 'apä',
    1092 => 'sañ',
    1093 => 'nih',
    1094 => 'hpa',
    1095 => 'rma',
    1096 => 'qhe',
    1097 => 'hep',
    1098 => 'uja',
    1099 => '_uj',
    1100 => 'ep_',
    1101 => 'aut',
    1102 => 'chp',
    1103 => 'arn',
    1104 => 'tut',
    1105 => 'ïki',
    1106 => 'sik',
    1107 => 'mäk',
    1108 => 'añj',
    1109 => 'ñta',
    1110 => 'uñt',
    1111 => 'ero',
    1112 => 'jki',
    1113 => 'fki',
    1114 => 'tt_',
    1115 => 'kfk',
    1116 => 'akf',
    1117 => 'llt',
    1118 => 'lta',
    1119 => 'ïsi',
    1120 => 'tt\'',
    1121 => 'eqp',
    1122 => 'uyi',
    1123 => 'uns',
    1124 => 'wki',
    1125 => 'qac',
    1126 => 'lít',
    1127 => 'jka',
    1128 => '\'in',
    1129 => 'alí',
    1130 => 'ujh',
    1131 => 'ruj',
    1132 => 'sjh',
    1133 => 'asj',
    1134 => '_\'i',
    1135 => 'rum',
    1136 => 'nqa',
    1137 => 'íta',
    1138 => 'awk',
    1139 => '_at',
    1140 => 'tip',
    1141 => 'ey',
    1142 => 'uq',
    1143 => 'tk',
    1144 => 'kw',
    1145 => 'pn',
    1146 => 'pu',
    1147 => 'os',
    1148 => 'jr',
    1149 => 'ij',
    1150 => 'jm',
    1151 => 'ah',
    1152 => '\'o',
    1153 => 'äc',
    1154 => 'ii',
    1155 => 'hh',
    1156 => 'íp',
    1157 => 'l_',
    1158 => 'rú',
    1159 => 'ú_',
    1160 => 'yu',
    1161 => 'yc',
    1162 => 'ñt',
    1163 => 'lp',
    1164 => 'äq',
    1165 => 'yo',
    1166 => 'hp',
    1167 => 'që',
    1168 => 'ís',
    1169 => 'ít',
    1170 => 'ñä',
    1171 => 'sm',
    1172 => 'kp',
    1173 => 'kc',
    1174 => 'ñk',
    1175 => 'lí',
    1176 => 'ua',
    1177 => 'eñ',
    1178 => 'p\'',
    1179 => 'sj',
    1180 => 'ëñ',
    1181 => 'yw',
    1182 => 'të',
    1183 => 'fk',
    1184 => 'wk',
    1185 => 'ët',
    1186 => 'má',
    1187 => 'ih',
    1188 => 'á_',
    1189 => 'pä',
    1190 => 'lt',
    1191 => 'jí',
    1192 => 'mr',
    1193 => 'jw',
    1194 => 'kf',
    1195 => 'pí',
    1196 => 'á',
    1197 => 'ú',
    1198 => 'f',
  ),
);
